src/main/java/org/littleshoot/proxy/impl/ProxyToServerConnection.java

```diff
@@ -528,6 +531,8 @@ private void respondWith(HttpObject httpObject) {
     private void connectAndWrite(HttpRequest initialRequest) {
         LOG.debug("Starting new connection to: {}", remoteAddress);
 
+        this.clientConnection.channel.attr(remoteAddressAttrKey).set(remoteAddress);
+
```

Comment:
There are 2 channels: `client to proxy` channel and `proxy to destination server` channel. 

The `proxy to server` channel is created from `client to proxy` channel and contains reference to it. This reference (to `client to proxy` channel) is passed to `mitm manager factory` from `ProxyToServerConnection` (when a connection to remote server is established). To select a proper `mitm manager` (with white listed hosts) we need to pass active server connection host to `client to proxy` channel so it is accessible from `ConditionallyInsecureMitmManagerFactory`